src/disk_io_thread_pool.cpp

```diff
@@ -158,7 +157,7 @@ namespace libtorrent {
 			; ++i)
 		{
 			// if this is the first thread started, start the reaper timer
-			if (m_threads.empty())
+			if (m_threads.empty() && !m_abort)
```

Comment:
just some defensive programming to make sure we don't reactivate the idle timer during shut down, as we may get stuck in `io_context::run()` waiting for it to trigger before exiting the last thread